Bottom: 6d0afef87db4378eee1a5ca8838b2084671d3c20
Top:    bc200acfdf479b5b5be9644d2add4bad3f5cb854
Author: Pavel Roskin <proski@gnu.org>
Date:   2013-11-05 12:36:08 -0500

ww

---

diff --git a/farm-config32 b/farm-config32
index cdfab13..7b526ca 100755
--- a/farm-config32
+++ b/farm-config32
@@ -6,6 +6,7 @@ set -e
 arch=i386
 ccflags=-m32
 
+error=
 for srctree in $(ls -d $farmsrc/* | sort -V); do
 	test -d $srctree/Documentation || continue
 	bintree=$farmbin/$(basename $srctree)-$arch
@@ -34,8 +35,17 @@ for srctree in $(ls -d $farmsrc/* | sort -V); do
 		date > $bintree/configured-$arch
 		echo -e "\e[32mOK\e[0m"
 	else
+		error=1
 		cd $srctree
 		mv $bintree $bintree-bad
 		echo -e "\e[31mERROR\e[0m"
 	fi
 done
+
+if test -z "$error"; then
+		echo -e "\e[32mAll OK\e[0m"
+		exit 0
+	else
+		echo -e "\e[31mErrors found\e[0m"
+		exit 1
+fi
diff --git a/farm-config64 b/farm-config64
index 38f80ab..4f21fd6 100755
--- a/farm-config64
+++ b/farm-config64
@@ -6,6 +6,7 @@ set -e
 arch=x86_64
 ccflags=-m64
 
+error=
 for srctree in $(ls -d $farmsrc/* | sort -V); do
 	test -d $srctree/Documentation || continue
 	bintree=$farmbin/$(basename $srctree)-$arch
@@ -34,8 +35,17 @@ for srctree in $(ls -d $farmsrc/* | sort -V); do
 		date > $bintree/configured-$arch
 		echo -e "\e[32mOK\e[0m"
 	else
+		error=1
 		cd $srctree
 		mv $bintree $bintree-bad
 		echo -e "\e[31mERROR\e[0m"
 	fi
 done
+
+if test -z "$error"; then
+		echo -e "\e[32mAll OK\e[0m"
+		exit 0
+	else
+		echo -e "\e[31mErrors found\e[0m"
+		exit 1
+fi
